Source/WebCore/bindings/js/JSDOMPromiseDeferred.h

```diff
@@ -202,6 +202,14 @@ class DeferredPromise : public DOMGuarded<JSC::JSPromise> {
     void whenSettled(Function<void()>&&);
     bool needsAbort() const { return m_needsAbort; }
 
+    void markAsHandled() const
+    {
+        ASSERT(deferred());
+        ASSERT(globalObject());
```

Comment:
I'm not sure how valuable this assert is. We're just going to dereference globalObject in `JSPromise::markAsHandled` anyway.

Reply:
Thanks for the review, looks like we can go with Tim's version.